src/TestableIO.System.IO.Abstractions.TestingHelpers/Events/OperationResponse.cs

```diff
@@ -0,0 +1,19 @@
+namespace System.IO.Abstractions.TestingHelpers.Events;
+
+/// <summary>
+/// Represents a response to a file system operation that can cancel or modify the operation.
+/// </summary>
+public class OperationResponse
+{
+    /// <summary>
+    /// Gets or sets a value indicating whether the operation should be cancelled.
+    /// Only applies to Before phase events.
+    /// </summary>
+    public bool Cancel { get; set; }
```

Comment:
Can't you combine the `Cancel` with the `Exception` property, as when using the `OperationResponse` both only throw an exception?

Something along the line:
```csharp
public bool Cancel()
{
    Exception = new OperationCanceledException("The Operation was cancelled.");
}
```